test/spec/FileSystem-test.js

```diff
@@ -56,10 +56,11 @@ define(function (require, exports, module) {
         }
 
         function readCallback() {
-            var callback = function (err, data, encoding, stat) {
+            var callback = function (err, data, encoding, preserveBOM, stat) {
```

Comment:
This test shouldn't be modified. The BOM flag is purely an internal config option and shouldn't be exposed. In case we want to expose this in future, it has to be a preference and that too only for write part.